packages/appcache-polyfill-sw/index.ts

```diff
@@ -0,0 +1,318 @@
+/*
+ Copyright 2019 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+/// <reference lib="webworker" />
+declare const self: ServiceWorkerGlobalScope;
+
+import {fetchWithFallback} from '../../lib/fetchWithFallback';
+import {longestMatchingPrefix} from '../../lib/longestMatchingPrefix';
+import * as storage from '../../lib/storageWithDefault';
+
+import {
+  ClientIdToHash,
+  Manifest,
+  ManifestURLToHashes,
+  PageURLToManifestURL,
+} from '../../lib/interfaces';
+
+async function getClientUrlForEvent(event: FetchEvent) {
```

Comment:
So, as far as I can tell, I did not end up needing to use `event.replacesClientId` or `event.resultingClientId` here. It seems like falling back to `event.request.url` or `event.request.referrer` was sufficient to get a client page URL when `clientId` isn't set.

Maybe I'm missing an edge case about why that's not sufficient, though?